### ISSUE TITLE
[Suggestion] Use a libre/open source forge instead of GitHub

### ISSUE THREAD

**Issue by tuxayo** [WolfireGames/overgrowth#72]
Because [Free Software Needs Free Tools](https://mako.cc/writing/hill-free_tools.html) and for not [reinforcing monopolies](https://nedbatchelder.com/blog/201405/github_monoculture.html) because even the most conservative capitalist view is that monopolies harm users.
Monopolies at the level of GitHub for software forges and oligopoly at the level of Microsoft.
It's also for the sake of avoiding the worst actors which Microsoft is a proven one whether it's for tax evasion, anti competitive practices, hindrance to the libre and open source software movements, [massive privacy breaching](https://en.wikipedia.org/wiki/Surveillance_capitalism), and even corruption for large [deals with education sector in France[fr]](https://www.anticor.org/2022/05/12/soupcons-de-favoritisme-dans-lattribution-dun-marche-de-leducation-nationale-anticor-saisit-le-pnf/).
More reading: https://en.wikipedia.org/wiki/Criticism_of_Microsoft

In short and simple: the digital world would be better if GitHub and Microsoft products would be less used in favor of libre ones hosted by respectable actors.

There are a lot of public Gitlab or Gitea instances and before exploring for a concrete one, what is the team take on the issue?

GitHub doesn't drive more contributions in itself, the project isn't significantly more discoverable because it's on GitHub. However account creation laziness is a thing that has a real impact which must be addressed. This is why a number of Gitlab and Gitea instances support auth via GitHub.

I think Gitlab or Gitea are good enough to not hinder contributions compared to GitHub and that any difference isn't really noticeable in absolute terms and even more so relatively when comparing to the tradeoff

**Comment by tuxayo** [comment]
Also: super exited to learn that overgrowth is libre! Thanks a lot! :heart:  A lot of games could do the same since keeping the assets proprietary doesn't change the business model

**Comment by autious** [comment]
I do personally share some of your concerns regarding Github and its ownership structure, and how that will affect the free open source world in the long run. In our case, our company is otherwise set up for using Github, and transferring the upstream repo to another platform would add complications to our routines. Right now in practice i don't see it being an issue, yet, and if there are signs that it might become one, it'll still be possible to transfer to a different site.

I'm not closing the door to this idea forever, but for the forseeable future, we're staying here for practical reasons.

These kinds of topics do get brought up internally however, and we do care.

------

You're welcome! It was fun to make it Libre!

From experience, very many games are sadly blocked from open sourcing their engines in this vein due to third party middlewares! Something i've become very aware of when working on making Overgrowth OSS ready. This is especially true for titles built in engines like Unity and Unreal, not due to the engines themselves, but how the surrounding infrastructure is set up for using asset-packs on different stores, where code and assets rarely allow redistribution in source form, and their code quickly gets woven into the core of the games.

**Comment by Conan-Kudo** [comment]
Lugaru (the predecessor to this game) is primarily hosted on GitLab.com. If we decide to move to GitLab in the future, we can certainly do that. Other platforms don't really have a good CI/CD story, which makes them hard to consider for our needs.
